lib/ae_mdw/db/contract.ex

```diff
@@ -317,19 +316,10 @@ defmodule AeMdw.Db.Contract do
 
   @spec aex9_search_contract(State.t(), pubkey(), integer()) :: map()
   def aex9_search_contract(state, account_pk, last_txi) do
-    gen_collect(
-      Model.Aex9AccountPresence,
-      {account_pk, last_txi, <<max_256bit_int()::256>>},
-      fn {acc_pk, _, _} -> acc_pk == account_pk end,
-      &State.prev(state, &1, &2),
-      fn -> %{} end,
-      fn {_, txi, ct_pk}, accum ->
-        Map.update(accum, ct_pk, [txi], fn txi_list ->
-          [txi | txi_list]
-        end)
-      end,
-      & &1
-    )
+    state
+    |> Collection.stream(Model.Aex9AccountPresence, {account_pk, last_txi, Util.min_int()})
+    |> Stream.take_while(&match?({^account_pk, _txi, _contract_pk}, &1))
+    |> Enum.group_by(fn {_account_pk, _txi, contract_pk} -> contract_pk end)
```

Comment:
need to return a mapping from `ct_pk` to `txi_list`

Reply:
That's what [`Enum.group_by/2`](https://hexdocs.pm/elixir/1.13/Enum.html#group_by/3) does

Reply:
nvm! I just realized that the values of the map weren't a txi list, my bad! fixed it now